From 7bda96c431321de1b0fe2b88ccb388ec4b0293dd Mon Sep 17 00:00:00 2001
From: Kazuhito Hagio <k-hagio-ab@nec.com>
Date: Tue, 19 Jan 2021 18:42:54 +0900
Subject: [PATCH 11/13] Fix "kmem -v" option on Linux 5.11-rc1 and later
 kernels

Fix the "kmem -v" option on Linux 5.11-rc1 and later kernels
that contains commit 96e2db456135db0cf2476b6890f1e8b2fdcf21eb
("mm/vmalloc: rework the drain logic").  Without the patch,
the option will display nothing or fail with the error message
"kmem: invalid kernel virtual address: <address>  type: "vmlist addr".

Signed-off-by: Kazuhito Hagio <k-hagio-ab@nec.com>
Signed-off-by: Lianbo Jiang <lijiang@redhat.com>
---
 memory.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/memory.c b/memory.c
index 0848097eb4f5..33b0ca7af977 100644
--- a/memory.c
+++ b/memory.c
@@ -403,8 +403,6 @@ vm_init(void)
 	    VALID_MEMBER(vmap_area_va_end) &&
 	    VALID_MEMBER(vmap_area_list) &&
 	    VALID_MEMBER(vmap_area_vm) &&
-	    (VALID_MEMBER(vmap_area_flags) || 
-		(OFFSET(vmap_area_vm) == MEMBER_OFFSET("vmap_area", "purge_list"))) &&
 	    kernel_symbol_exists("vmap_area_list"))
 		vt->flags |= USE_VMAP_AREA;
 
-- 
2.17.1

